### PR TITLE
Add semantic headings

### LINKED ISSUE #10427: Add semantic headings to all pages
The lack of semantic headings (`<h1>`, etc.) on some pages has been called out as a potential issue by accessibility reviewers. We should add such headings to ensure that PrairieLearn is meeting accessibility requirements.

On many pages, we use cards that already have headings in the header, they just don't necessarily use `<h1>`/etc. tags. It should be relatively easy to change those all to semantic headings; we just might want some custom CSS for that.

On pages that might not use cards but still need a title for screen readers, we can potentially use screenreader-only titles via Bootstrap's `sr-only` class. This will ensure that folks using screen readers can easily navigate the page, without cluttering it up with unnecessary titles for others.

### PR DESCRIPTION
**Pull request by mylesw27**
Closes #10427. This PR adds `<h1>` headings to all pages that do not have one. This will help us meet accessibility requirements by having a heading that is readable by screen readers on all pages. For most pages here, I have updated the card header to be the `<h1>` title for the page. Pages that did not have a clear heading or title, I have added a heading with `.sr-only`. The goal here is to have very little impact on the visual presentation of the pages while adding these headings.

### CODE REVIEW

File: apps/prairielearn/src/components/InstructorInfoPanel.html.ts [hunk 1]
```diff
@@ -97,7 +97,7 @@ export function InstructorInfoPanel({
 
 function StaffUserInfo({ user }: { user: User }) {
   return html`
-    <h5 class="card-title">Staff user:</h5>
+    <h2 class="card-title h5">Staff user:</h2>
```
Comment (mylesw27) [inline apps/prairielearn/src/components/InstructorInfoPanel.html.ts:100]: This change and below were done to preserve the order of headings on the page.

File: apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts [hunk 2]
```diff
@@ -177,7 +177,7 @@ export function AdministratorQuery({
         <main id="content" class="container-fluid">
           <div class="card mb-4">
             <div class="card-header bg-primary text-white d-flex align-items-center">
-              <span class="font-weight-bold text-monospace">${sqlFilename}</span>
+              <h1 class="h6 font-weight-bold text-monospace mb-0">${sqlFilename}</h1>
```
Comment (mylesw27) [inline apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts:180]: Using the sql filename here for the title. It works but it might be strange reading a filename instead here. Let me know if you want me to use something else more standard like "Admin Query" or something here.

File: apps/prairielearn/src/pages/administratorFeatures/administratorFeatures.html.ts [hunk 3]
```diff
@@ -47,7 +47,7 @@ export function AdministratorFeatures({
         })}
         <main id="content" class="container">
           <div class="card mb-4">
-            <div class="card-header bg-primary text-white">Features</div>
+            <h1 class="card-header bg-primary text-white h6 font-weight-normal">Features</h1>
```
Comment (mylesw27) [inline apps/prairielearn/src/pages/administratorFeatures/administratorFeatures.html.ts:50]: For card headers without additional buttons, I updated the `<div>` to be the `<h1>` heading as to reduce the amount of elements used here. Let me know if you would prefer the `<h1>` to be a child of the `<div>` for these instead.

File: apps/prairielearn/src/components/QuestionContainer.html.ts [hunk 4]
```diff
@@ -617,11 +617,13 @@ function QuestionPanel({
...
-          question,
-          questionNumber: instance_question_info?.question_number,
-        })}
+        <h1 class="h6 font-weight-normal mb-0">
+          ${QuestionTitle({
+            questionContext,
+            question,
+            questionNumber: instance_question_info?.question_number,
+          })}
+        </h1>
```
Comment (mylesw27) [inline apps/prairielearn/src/components/QuestionContainer.html.ts:626]: For pages where `QuestionContainer` was being used, it made sense that this would be the the `<h1>` heading for those pages so I updated here instead of adding to the page html.

**Review by mylesw27 — COMMENTED** [review]
Most of the changes here were pretty straightforward. I left a few comments here to help provide some clarity on anything that may have been strange. Let me know if there are any places to improve!

File: apps/prairielearn/src/pages/administratorFeatures/administratorFeatures.html.ts (hunk 3 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/administratorFeatures/administratorFeatures.html.ts:50]: I would indeed prefer to nest `<h1>` inside the `div.card-header`. This will make sure that we don't accidentally end up announcing action buttons as part of the heading if we add them later.

File: apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts:180]: This is an admin-only page so this matters much less than user-facing ones, but you might consider changing this to something like `Query: <span class="text-monospace">${sqlFilename}</span>` so that it's at least announced as a query and not just a filename.

File: apps/prairielearn/src/pages/administratorFeatures/administratorFeatures.html.ts (hunk 3 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/administratorFeatures/administratorFeatures.html.ts]: Sorry, I should have done this on the first go around. Updated to nesting the `<h1>` in [2d90727](https://github.com/PrairieLearn/PrairieLearn/pull/10440/commits/2d907275890a1dae887227f7e0d2c248110fa04e).

File: apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts (hunk 2 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts]: That sounds good. Updated to this in [32a4353](https://github.com/PrairieLearn/PrairieLearn/pull/10440/commits/32a4353c31f3a6e43f583964f80031e262180873).

File: apps/prairielearn/src/components/Modal.html.ts [hunk 5]
```diff
@@ -36,7 +36,7 @@ export function Modal({
             ${formAction ? html`action="${formAction}"` : ''}
           >
             <div class="modal-header">
-              <h4 class="modal-title" id=${titleId}>${title}</h4>
+              <h2 class="modal-title h4" id=${titleId}>${title}</h2>
```
Comment (nwalters512) [inline apps/prairielearn/src/components/Modal.html.ts:39]: Nice catch!

File: apps/prairielearn/src/pages/courseSyncs/courseSyncs.html.ts [hunk 6]
```diff
@@ -57,6 +57,7 @@ export function CourseSyncs({
         </script>
         ${renderEjs(import.meta.url, "<%- include('../partials/navbar'); %>", resLocals)}
         <main id="content" class="container-fluid">
+          <h1 class="sr-only">Course Sync</h1>
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/courseSyncs/courseSyncs.html.ts:60]: On pages like this where we have multiple cards, my inclination would be to use `<h2>` for card titles to give screen readers additional navigational landmarks. I won't block on this and I'd be happy to save this for another PR.

File: apps/prairielearn/src/components/QuestionContainer.html.ts (hunk 4 above)
Comment (nwalters512) [reply apps/prairielearn/src/components/QuestionContainer.html.ts:626]: I think we could potentially make the question page much nicer for screen readers by adding a lot more navigational landmarks, e.g. separate headings for the question panel, correct answer panel, all submission panels, and so on. Let's save this for another PR though so we can discuss and review that in isolation. Would you like to make an issue for this?

File: apps/prairielearn/src/pages/instructorIssues/instructorIssues.html.ts [hunk 7]
```diff
@@ -408,20 +408,20 @@ function FilterHelpModal() {
           </tr>
         </tbody>
       </table>
-      <h4>Full-text search</h4>
+      <h2 class="h4">Full-text search</h2>
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorIssues/instructorIssues.html.ts:411]: Given that modals now use an `<h2>` for their title, an `<h3>` might be more appropriate here.

File: apps/prairielearn/src/pages/instructorFileBrowser/instructorFileBrowser.html.ts [hunk 8]
```diff
@@ -196,6 +198,7 @@ export function InstructorFileBrowser({
         ${renderEjs(import.meta.url, "<%- include('../partials/navbar'); %>", resLocals)}
         <main id="content" class="container-fluid">
           ${syncErrorsAndWarnings}
+          <h1 class="sr-only">File Browser</h1>
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorFileBrowser/instructorFileBrowser.html.ts:201]: Nit, probably use the same heading for both authorized and non-authorized views?

File: apps/prairielearn/src/pages/instructorInstanceAdminAccess/instructorInstanceAdminAccess.html.ts [hunk 9]
```diff
@@ -38,7 +38,9 @@ export function InstructorInstanceAdminAccess({
 
           <div class="card mb-4">
             <div class="card-header bg-primary text-white d-flex align-items-center">
-              ${course_instance.long_name} course instance access rules
+              <h1 class="h6 font-weight-normal mb-0">
+                ${course_instance.long_name} course instance access rules
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorInstanceAdminAccess/instructorInstanceAdminAccess.html.ts:42]: This isn't a problem with this PR, but we're kinda inconsistent in how these headings are constructed. I see three variants:

- "LONG NAME course instance access rules" (this page)
- "LTI configuration" and similar
- "Course instance SHORT NAME" (the settings page)

Not requesting a change here, but it'd be nice to be more consistent for folks who are using screen readers. Maybe another issue to track this?

File: apps/prairielearn/src/pages/courseSyncs/courseSyncs.html.ts (hunk 6 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/courseSyncs/courseSyncs.html.ts:60]: Makes sense! I have opened issue #10446 and mentioned this as part of that issue.

File: apps/prairielearn/src/components/QuestionContainer.html.ts (hunk 4 above)
Comment (mylesw27) [reply apps/prairielearn/src/components/QuestionContainer.html.ts:626]: That is a great idea. I have now opened issue #10446 to track this.

File: apps/prairielearn/src/pages/instructorIssues/instructorIssues.html.ts (hunk 7 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorIssues/instructorIssues.html.ts]: Updated in [7d5e2bc](https://github.com/PrairieLearn/PrairieLearn/pull/10440/commits/7d5e2bc50e687b8411d196edb8040872f0b802dd).

File: apps/prairielearn/src/pages/instructorFileBrowser/instructorFileBrowser.html.ts (hunk 8 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorFileBrowser/instructorFileBrowser.html.ts:200]: Updated in [af07a83](https://github.com/PrairieLearn/PrairieLearn/pull/10440/commits/af07a83e2931531fae4210753b6f9e267a4b0567).

File: apps/prairielearn/src/pages/instructorInstanceAdminAccess/instructorInstanceAdminAccess.html.ts (hunk 9 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorInstanceAdminAccess/instructorInstanceAdminAccess.html.ts:42]: I agree. Opened issue #10447 to track this.

File: apps/prairielearn/src/pages/instructorFileBrowser/instructorFileBrowser.html.ts (hunk 8 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorFileBrowser/instructorFileBrowser.html.ts:200]: I would in fact go the other way and standardize on "Files" (that is, keep the existing visual title, and make the screenreader-only title match it).

Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorFileBrowser/instructorFileBrowser.html.ts]: Did you mean something like this, [38fde3a](https://github.com/PrairieLearn/PrairieLearn/pull/10440/commits/38fde3aac2c083893cff8c78824c85fda866dfec)?

Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorFileBrowser/instructorFileBrowser.html.ts]: Exactly!

**Review by nwalters512 — APPROVED** [review]
